### PR TITLE
Add support for building Mac Catalyst apps

### PR DESCRIPTION
**Pull request by foyoodo**
Relates to #7122

### Short description 📝

Add support for building Mac Catalyst apps.

### How to test the changes locally 🧐

> Include a set of steps for the reviewer to test the changes locally (see [the documentation](https://docs.tuist.io/contributors/get-started) for reference).

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift [hunk 1]
```diff
@@ -158,6 +158,10 @@ public final class DefaultSettingsProvider: DefaultSettingsProviding {
             )
         }
 
+        if target.supportsCatalyst {
+            settings.overlay(with: ["CODE_SIGN_IDENTITY": "-"], for: .macOS)
+        }
```
Comment (fortmarek) [inline Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift:163]: Why is this needed?

File: Tests/TuistGeneratorAcceptanceTests/GenerateAcceptanceTests.swift [hunk 2]
```diff
@@ -917,6 +917,20 @@ final class GenerateAcceptanceTestiOSAppWithWeaklyLinkedFramework: TuistAcceptan
     }
 }
 
+final class GenerateAcceptanceTestiOSAppWithCatalyst: TuistAcceptanceTestCase {
+    func test_ios_app_with_catalyst() async throws {
+        try await setUpFixture(.iosAppWithCatalyst)
+        try await run(GenerateCommand.self)
+        try await run(BuildCommand.self, "App", "--platform", "macos")
```
Comment (fortmarek) [inline Tests/TuistGeneratorAcceptanceTests/GenerateAcceptanceTests.swift:924]:
```suggestion
        try await run(BuildCommand.self, "App", "--platform", "macos")
        try await run(BuildCommand.self, "App", "--platform", "ios")
```

nit: I'd also build the iOS platform since it's an _iOS_ app with catalyst

File: fixtures/ios_app_with_catalyst/App/Resources/Assets.xcassets/AccentColor.colorset/Contents.json
Comment (fortmarek) [inline fixtures/ios_app_with_catalyst/App/Resources/Assets.xcassets/AccentColor.colorset/Contents.json:1]: I'd completely remove the `Resources` directory as we don't need it to keep the fixture more focused to what it's actually testing.

File: Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift (hunk 1 above)
Comment (foyoodo) [reply Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift:163]: This approach sets the default `CODE_SIGN_IDENTITY` for the macOS destination to 'Sign to Run Locally'. I noticed similar logic in `XcodeProj`, which allows consumers to run the project directly without specifying `CODE_SIGN_IDENTITY`. Do you think this logic should be moved into `XcodeProj`?

Comment (fortmarek) [reply Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift:163]: No, I think it's fine to keep it here. Thanks for the explanation!

Comment (fortmarek) [reply Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift:163]: Can you add a comment to the source code with the explanation, so it's obvious to anybody else who reads that piece in the future?

Comment (foyoodo) [reply Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift:163]: Sure, I’ll add a comment to clarify the purpose for future readers.

**Comment by pepicrft** [comment]
@all-contributors add @foyoodo for code

**Review by pepicrft — APPROVED** [review]
I addressed the remaining comments, so the PR should be ready to merge if CI is happy :)
